src/main/java/mops/gruppen2/domain/model/group/Group.java

```diff
@@ -0,0 +1,313 @@
+package mops.gruppen2.domain.model.group;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+import mops.gruppen2.domain.exception.BadArgumentException;
+import mops.gruppen2.domain.exception.GroupFullException;
+import mops.gruppen2.domain.exception.IdMismatchException;
+import mops.gruppen2.domain.exception.LastAdminException;
+import mops.gruppen2.domain.exception.NoAccessException;
+import mops.gruppen2.domain.exception.UserAlreadyExistsException;
+import mops.gruppen2.domain.exception.UserNotFoundException;
+import mops.gruppen2.domain.helper.CommonHelper;
+import mops.gruppen2.domain.helper.ValidationHelper;
+import mops.gruppen2.domain.model.group.wrapper.Body;
+import mops.gruppen2.domain.model.group.wrapper.Description;
+import mops.gruppen2.domain.model.group.wrapper.Limit;
+import mops.gruppen2.domain.model.group.wrapper.Link;
+import mops.gruppen2.domain.model.group.wrapper.Parent;
+import mops.gruppen2.domain.model.group.wrapper.Title;
+
+import javax.validation.Valid;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * Repräsentiert den aggregierten Zustand einer Gruppe.
+ *
+ * <p>
+ * Muss beim Start gesetzt werden: groupid, meta
+ */
+@Log4j2
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+public class Group {
+
+    // Metainformationen
+    @EqualsAndHashCode.Include
+    private UUID groupid;
+
+    @Getter
+    private Type type = Type.PRIVATE;
+
+    private Parent parent = Parent.EMPTY();
+
+    private Limit limit = Limit.DEFAULT(); // Add initial user
+
+    private Link link = Link.RANDOM();
+
+    private GroupMeta meta = GroupMeta.EMPTY();
+
+    private GroupOptions options = GroupOptions.DEFAULT();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'options' could be replaced by a local variable.](https://app.codacy.com/manual/tobi_2/gruppenfindung/pullRequest?prid=5356872)